.functional_ci_setup.py

```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/env python
+import base64
+import binascii
+import os
+import yaml
+
+from os import environ as env
+from configparser import ConfigParser
+
+
+def write_aws_creds():
+    aws_profile = env['AWS_PROFILE']
+    # Write the credentials file
+    creds = ConfigParser()
+    creds[aws_profile] = dict(
+        aws_access_key_id=env['AWS_ACCESS_KEY_ID'],
+        aws_secret_access_key=env['AWS_SECRET_ACCESS_KEY'],
+    )
+    creds_path = env['AWS_SHARED_CREDENTIALS_FILE']
+    os.makedirs(
+        os.path.dirname(creds_path),
+        exist_ok=True,
+    )
+    with open(creds_path, 'w') as creds_file:
+        creds.write(creds_file)
+
+    # Write the config file
+    conf = ConfigParser()
+    conf[aws_profile] = dict(
+        region=env['AWS_REGION'],
+        output='text',
+    )
+    conf_path = env['AWS_CONFIG_FILE']
+    os.makedirs(
+        os.path.dirname(conf_path),
+        exist_ok=True,
+    )
+    with open(conf_path, 'w') as conf_file:
+        conf.write(conf_file)
+
+
+def write_pull_secret():
+    secret_dir = os.path.join(env['WORKSPACE'], 'data')
+    os.makedirs(secret_dir, exist_ok=True)
+    secret = env['PULL_SECRET']
```

Comment:
Not sure what you mean here. In Jenkins, this will just be a JSON string. In GitLab CI, it *must* be base64-encoded, as that's a requirement of theirs for secrets beyond a certain length.

Reply:
If it's cause of the length you can directly use jenkins credentials as secret file, you can directly upload this file and use it an copy it to proper location you need from the pipeline directly

Reply:
Decoding doesn't happen at all when this is run in Jenkins, though.